documentation/collection/integration/sveltekit/basics/sign-out-users.md

```diff
@@ -3,23 +3,37 @@ _order: 4
 title: "Sign out users"
 ---
 
-You can invalidate the current session and remove the session cookies from the client using [`signOut()`](/sveltekit/api-reference/client-api#signout), exported by `@lucia-auth/sveltekit/client`. This can only be called in the client (`+page.svelte`) as it relies on the native `fetch` API.
+An action that handles log out should be created and a sign out request should be sent using forms. It can alternatively be an API endpoint.
 
-```ts
-import { signOut } from "@lucia-auth/sveltekit/client";
+## Handle Request
 
-await signOut();
-```
+Sign out requests should be handled by POST. **Make sure to invalidate the current session** and delete the cookie by setting the current session to `null`.
 
-You can use [`invalidateAll()`](https://kit.svelte.dev/docs/modules#$app-navigation-invalidateall) to re-run all load functions and let the load functions handle the redirect.
+```ts
+import { type Actions, fail } from "@sveltejs/kit";
+import type { PageServerLoad } from "./$types";
+import { auth } from "$lib/server/lucia";
+
+export const actions: Actions = {
+	default: async ({ locals }) => {
+		const session = await locals.validate();
+		if (!session) throw fail(401);
+		await auth.invalidateSession(session.sessionId); // invalidate session
+		locals.setSession(null); // remove cookie
+	}
+};
+```
 
-## Example
+## Submit request
 
-This will redirect the user to /login on successful sign out.
+Instead of a sign out button, add a form. Authenticated users will be signed out on a successful submission.
 
-```ts
-import { signOut } from "@lucia-auth/sveltekit/client";
+```svelte
+<script lang="ts">
+	import { enhance } from "$app/forms";
+</script>
 
-await signOut();
-invalidateAll();
+<form use:enhance method="post">
```

Comment:
The only thing I'm a bit wary about is the lack of `action=/some/route` in these example forms. People tend to place the sign-in and sign-out buttons in the layout, which will be accessible to any child route. If the action lives in the route `/` and they press the sign-out button while they are in the `/some/other/route`, the post will fail

Reply:
I'm guessing SvelteKit `use:enhance` works with forms with `action` pointing to `/some/other/route`?

Reply:
It's not so much as a `use:enhance` issue, but it's just how forms work in general. If the action is accessible from pages that aren't where the action exists, we _need_ to set the form's `action` attribute to point to the action's route. 

Say we have a default action that exists in the file `/+page.server.ts`, but the form is accessible from `/some/other/route/+page.svelte` (because the form lives in a `+layout.svelte` file), we have to set the action to: 

`<form method="post" action="/" use:enhance>`.

In other words, a form with no `action` attribute will only work when the form lives in the route that the action is declared.

Now this could totally just not be our problem and we should assume that the user should already understand how form actions work in SvelteKit. The idea of this being a problem was just a small thought that crossed my mind. It's not really a big deal.

Reply:
I know what you mean, was just unsure if SvelteKit allowed it. Will make a minor update to the docs